### PR TITLE
Various minor updates, v4 metadata changes.

### PR DESCRIPTION
**Pull request by evoskuil**
Please do not merge until corresponding changes in other repos are ready.

### CODE REVIEW

File: builds/msvc/vs2013/libbitcoin-examples/libbitcoin-examples.vcxproj.filters [hunk 1]
```diff
@@ -8,17 +8,11 @@
...
     </Filter>
     <Filter Include="src">
       <UniqueIdentifier>{B726DF7D-6D1D-48FB-0000-000000000000}</UniqueIdentifier>
     </Filter>
-    <Filter Include="src\.deps">
-      <UniqueIdentifier>{B726DF7D-6D1D-48FB-0000-000000000001}</UniqueIdentifier>
-    </Filter>
-    <Filter Include="src\.libs">
-      <UniqueIdentifier>{B726DF7D-6D1D-48FB-0000-000000000002}</UniqueIdentifier>
-    </Filter>
```
Comment (evoskuil) [inline builds/msvc/vs2013/libbitcoin-examples/libbitcoin-examples.vcxproj.filters]: @pmienk Could you update the build system to prevent this?

Comment (pmienk) [reply builds/msvc/vs2013/libbitcoin-examples/libbitcoin-examples.vcxproj.filters]: Will look at it now.  Looks like you'd like to exclude anything 'hidden' - from both file and directory listings?

Comment (evoskuil) [reply builds/msvc/vs2013/libbitcoin-examples/libbitcoin-examples.vcxproj.filters]: There are a few hidden files that we use, but no hidden directories. Not sure if the hidden files are relevant to the source listings though.

Comment (pmienk) [reply builds/msvc/vs2013/libbitcoin-examples/libbitcoin-examples.vcxproj.filters]: I had some errand .gitignore show up in one file listing, so I'm hoping we can ignore all of them or I can track it down and remove it.  Its contents can move into the top-level one.  But I'll rewrite the discovery function to allow for directory and file ignoring as separate parameters.

Comment (pmienk) [reply builds/msvc/vs2013/libbitcoin-examples/libbitcoin-examples.vcxproj.filters]: Addressed by libbitcoin/libbitcoin-build#164.  I did not see any use of hidden files and so have currently excluded them.  Also finally pulled the file discovery code into utilities (it's also used in my in-progress cmake build)

**Pull request merged**